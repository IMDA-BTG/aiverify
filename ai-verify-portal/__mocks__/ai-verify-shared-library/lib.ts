import { createContext } from 'react';

export function getComponents() {
  return {};
}

export function parseRJSFSchema() {
  return {
    sensitive_feature: {
      items: {},
    },
  };
}

export type InputDataContextType = {
  meta: any;
  data: any;
  onChangeData?: (key: string, value: any) => void;
};

export const InputDataContext = createContext<InputDataContextType>({
  meta: {},
  data: {},
});
