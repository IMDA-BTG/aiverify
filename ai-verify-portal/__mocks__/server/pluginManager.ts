import { projectFlow_mockPlugins } from '__mocks__/mockPlugins';

export function getPlugins() {
  return projectFlow_mockPlugins;
}
