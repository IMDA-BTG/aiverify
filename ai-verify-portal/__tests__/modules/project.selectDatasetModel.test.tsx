import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { mockDomMatrix, silentConsoleLogs } from '__mocks__/mockGlobals';
import { useEffect, useState } from 'react';
import ProjectCreatePage from 'pages/project/create';
import { getPlugins } from 'server/pluginManager';
import { MockProviders } from '__mocks__/mockProviders';
import PluginManagerType from 'src/types/pluginManager.interface';
import {
  MOCK_ALGO_ID,
  MOCK_DATE_WIDGET_1,
  MOCK_IBLOCK_ID,
  mockGqlDataE2E,
} from '__mocks__/mockGqlResponse';
import {
  inputBlockMdxCompBundleResponse,
  inputBlockMdxSummaryBundleResponse,
  widgetMdxBundleResponse,
} from '__mocks__/mockPlugins';
import { DataGridProps } from '@mui/x-data-grid';

jest.mock('@mui/x-data-grid', () => {
  const { DataGrid } = jest.requireActual('@mui/x-data-grid');
  return {
    ...jest.requireActual('@mui/x-data-grid'),
    DataGrid: (props: DataGridProps) => {
      return <DataGrid {...props} autoPageSize={false} />; //autoPageSize={false} to overcome rows not rendered in jsdom
    },
  };
});

const GRID_LAYOUT_CLASSNAME = '.react-grid-layout';
const GRID_ITEM_CLASSNAME = '.react-grid-item';

function ProjectCreatePageWrapper() {
  const [plugins, setPlugins] = useState<PluginManagerType>();
  async function fetchPlugins() {
    const result = await getPlugins();
    return result;
  }
  useEffect(() => {
    (async () => {
      const fetchedPlugins = await fetchPlugins();
      setPlugins(fetchedPlugins);
    })();
  }, []);

  return (
    <div>
      <div id="aivModal" />
      <ProjectCreatePage pluginManager={plugins as PluginManagerType} />
    </div>
  );
}

const fetchSpy = jest.spyOn(global, 'fetch');

describe('Project Flow - Select Dataset And Model', () => {
  beforeAll(() => {
    jest.useFakeTimers({ advanceTimers: true });
    silentConsoleLogs();
    mockDomMatrix();
  });

  afterEach(() => {
    fetchSpy.mockRestore();
  });

  describe('Datasets and Model Selection Screen', () => {
    it('should render Datasets / Model selection screen', async () => {
      // const fetchSpy = jest.spyOn(global, 'fetch');
      fetchSpy
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(widgetMdxBundleResponse),
        } as unknown as Response)
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(inputBlockMdxSummaryBundleResponse),
        } as unknown as Response);
      const user = userEvent.setup();
      const { container } = render(
        <MockProviders apolloMocks={mockGqlDataE2E}>
          <ProjectCreatePageWrapper />
        </MockProviders>
      );
      const projNameInput: HTMLInputElement =
        await screen.findByPlaceholderText(
          /^Enter name of this project e.g. Credit Scoring Model Tests$/i
        );
      expect(projNameInput).not.toBeNull();

      await user.type(projNameInput, 'Test Project');
      jest.runAllTimers();
      const nextBtn = await screen.findByText(/^Next$/i);
      await user.click(nextBtn.parentElement as HTMLDivElement);
      await screen.findByText(/^Select Report Template$/i);
      const blankCard = await screen.findByText(/^Blank Canvas$/i);
      await user.click(blankCard);
      await waitFor(() => {
        expect(container.querySelector('.card__highlighted')).toBeDefined();
      });
      await user.click(nextBtn);
      await screen.queryByText(/^Design Report$/i);
      await screen.findByText(/^Interpretation \(MAE\)$/i);
      const gridLayoutArea = container.querySelector(
        GRID_LAYOUT_CLASSNAME
      ) as HTMLDivElement;
      expect(gridLayoutArea.children.length).toBe(0);
      const maeInterWidget = container.querySelector(
        '[data-test-id="draggableWidget-aiverify.test.mock_test_plugin1:widget_cid_1"]'
      );

      // stub date.now to return mock date, which is used as a 'key' value when widget added to canvas
      const dateNowStub = jest.fn(() => MOCK_DATE_WIDGET_1);
      const actualDateNowFn = Date.now.bind(global.Date);
      global.Date.now = dateNowStub;

      fireEvent.dragStart(maeInterWidget as Element, {
        dataTransfer: { setData: (_str: string) => undefined },
        clientX: 100,
        clientY: 200,
        screenX: -1500,
        screenY: 200,
        x: 100,
        y: 200,
      });

      fireEvent.dragOver(gridLayoutArea as Element, {
        clientX: 100,
        clientY: 200,
        screenX: 600,
        screenY: 200,
        x: 200,
        y: 200,
      });

      await waitFor(() => {
        expect(container.querySelector(GRID_ITEM_CLASSNAME)).not.toBeNull();
      });

      fireEvent.drop(gridLayoutArea as Element, {
        clientX: 100,
        clientY: 200,
        screenX: 600,
        screenY: 200,
        x: 300,
        y: 500,
      });

      //widget should be on canvas
      await waitFor(() => {
        expect(container.querySelector('.widgetContainer')).not.toBeNull();
      });

      //reset date stub
      global.Date.now = actualDateNowFn;

      const saveBtn = container.querySelector(
        'svg[data-testid="SaveIcon"]'
      ) as HTMLElement;
      await user.click(saveBtn);
      await screen.findByText(/^Project saved$/i);

      await user.click(await screen.findByText(/^Next$/i));
      await screen.findByText(
        /^Select the Datasets and AI Model to be tested$/i
      );
      expect(container.querySelector('.layoutContentArea')).toMatchSnapshot(
        'Datasets/Model-Selection-Screen'
      ); // snapshot without header, because header has dynamic autosave time display
    });

    it('should select Datasets, Ground truth, capture Test arguments, Input Blocks and call generate report endpoints', async () => {
      // Note: /api/bundler/summary is being called 5 times on select user input screen in unit test. On browser it's called twice. Consider using react.query for chaching strategy
      fetchSpy
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(widgetMdxBundleResponse),
        } as unknown as Response)
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(inputBlockMdxSummaryBundleResponse), // mocked 5 times. here and next few ones
        } as unknown as Response)
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(inputBlockMdxSummaryBundleResponse), // same as previous
        } as unknown as Response)
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(inputBlockMdxSummaryBundleResponse), // same as previous
        } as unknown as Response)
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(inputBlockMdxSummaryBundleResponse), // same as previous
        } as unknown as Response)
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(inputBlockMdxSummaryBundleResponse), // same as previous
        } as unknown as Response)
        .mockResolvedValueOnce({
          status: 200,
          json: () => Promise.resolve(inputBlockMdxCompBundleResponse),
        } as unknown as Response);
      const user = userEvent.setup();
      const { container } = render(
        <MockProviders apolloMocks={mockGqlDataE2E}>
          <ProjectCreatePageWrapper />
        </MockProviders>
      );
      const projNameInput: HTMLInputElement =
        await screen.findByPlaceholderText(
          /^Enter name of this project e.g. Credit Scoring Model Tests$/i
        );
      await user.type(projNameInput, 'Test Project');
      jest.runAllTimers();
      const nextBtn = await screen.findByText(/^Next$/i);
      await user.click(nextBtn.parentElement as HTMLDivElement);
      await screen.findByText(/^Select Report Template$/i);
      const blankCard = await screen.findByText(/^Blank Canvas$/i);
      await user.click(blankCard);
      await waitFor(() => {
        expect(container.querySelector('.card__highlighted')).toBeDefined();
      });
      await user.click(nextBtn);
      await screen.queryByText(/^Design Report$/i);
      await screen.findByText(/^Interpretation \(MAE\)$/i);
      const gridLayoutArea = container.querySelector(
        GRID_LAYOUT_CLASSNAME
      ) as HTMLDivElement;
      expect(gridLayoutArea.children.length).toBe(0);
      const maeInterWidget = container.querySelector(
        '[data-test-id="draggableWidget-aiverify.test.mock_test_plugin1:widget_cid_1"]'
      );

      // stub date down to return mock date, which is used as a 'key' value when widget added to canvas
      const dateNowStub = jest.fn(() => MOCK_DATE_WIDGET_1);
      const actualDateNowFn = Date.now.bind(global.Date);
      global.Date.now = dateNowStub;

      fireEvent.dragStart(maeInterWidget as Element, {
        dataTransfer: { setData: (_str: string) => undefined },
        clientX: 100,
        clientY: 200,
        screenX: -1500,
        screenY: 200,
        x: 100,
        y: 200,
      });

      fireEvent.dragOver(gridLayoutArea as Element, {
        clientX: 100,
        clientY: 200,
        screenX: 600,
        screenY: 200,
        x: 200,
        y: 200,
      });

      await waitFor(() => {
        expect(container.querySelector(GRID_ITEM_CLASSNAME)).not.toBeNull();
      });

      fireEvent.drop(gridLayoutArea as Element, {
        clientX: 100,
        clientY: 200,
        screenX: 600,
        screenY: 200,
        x: 300,
        y: 500,
      });

      //widget should be on canvas
      await waitFor(() => {
        expect(container.querySelector('.widgetContainer')).not.toBeNull();
      });

      global.Date.now = actualDateNowFn;

      const counterChips = Array.from(
        container.querySelectorAll('.testsCountChip')
      );
      expect(counterChips[0].textContent).toBe('1'); //tests to run
      expect(counterChips[1].textContent).toBe('1'); //input blocks

      await user.click(await screen.findByText(/^Next$/i));
      await screen.findByText(
        /^Select the Datasets and AI Model to be tested$/i
      );
      const chooseDatasetBtns1 = await screen.findAllByText(
        /^Choose Dataset$/i
      );
      expect(chooseDatasetBtns1.length).toBe(2);

      // expect(container.querySelector('#inputsLeftpanel')).toMatchSnapshot(
      //   'leftPanel progress tracker BEFORE valid inputs'
      // );

      // Select Dataset (file picker)
      await user.click(chooseDatasetBtns1[0]);
      await screen.findByText(/^Choose the Dataset$/i);
      const datasetFile = await screen.findByText(
        /^pickle_pandas_tabular_compas_testing\.sav$/i
      );
      await waitFor(async () => {
        expect(
          container.querySelectorAll('div[aria-rowcount="3"]').length
        ).toBe(1);
      });
      expect(container.querySelectorAll('div[role="row"]').length).toBe(3);

      await user.click(datasetFile);
      await user.click(await screen.findByText(/^Use Dataset$/i));
      await screen.findByText(
        /^Select the Datasets and AI Model to be tested$/i
      );
      await screen.findByText(/^pickle_pandas_tabular_compas_testing\.sav$/i);
      const chooseDatasetBtns2 = await screen.findAllByText('Choose Dataset');
      expect(chooseDatasetBtns2.length).toBe(1);

      // Select Ground Truth Dataset (file picker)
      await user.click(chooseDatasetBtns2[0]);
      await screen.findByText(/^Choose Ground Truth Dataset$/i);
      await waitFor(async () => {
        expect(
          container.querySelectorAll('div[aria-rowcount="3"]').length
        ).toBe(1);
      });
      expect(container.querySelectorAll('div[role="row"]').length).toBe(3);
      const groundTruthFile = await screen.findByText(
        /^pickle_pandas_tabular_compas_testing\.sav$/i
      );
      await user.click(groundTruthFile);
      await user.click(await screen.findByText(/^Use Dataset$/i));
      await screen.findByText(
        /^Select the Datasets and AI Model to be tested$/i
      );

      // Select Ground Truth column
      await screen.findByText(/^Select Ground Truth:$/i);
      const groundTruthSelectInput = container.querySelector(
        '.inputCard #groundTruthSelect'
      ) as HTMLElement;
      await user.click(groundTruthSelectInput);
      await screen.findByText(/^Please Select$/i);
      await screen.findByText(/^age_cat_cat$/i);
      await screen.findByText(/^sex_code$/i);
      await screen.findByText(/^race_code$/i);
      await screen.findByText(/^priors_count$/i);
      await screen.findByText(/^c_charge_degree_cat$/i);
      const selected = await screen.findByText(/^two_year_recid$/i);
      await user.click(selected);
      await waitFor(async () => {
        expect(await screen.queryByText(/^Please Select$/i)).toBeNull();
      });
      expect(await screen.queryByText(/^two_year_recid$/i)).not.toBeNull();

      // Select Model
      await user.click(await screen.findByText(/^Choose Model$/i));
      await screen.findByText(/^Choose the Model$/i);
      const modelFile = await screen.findByText(
        /^pickle_scikit_bc_compas.sav$/i
      );
      await user.click(modelFile);
      await user.click(await screen.findByText(/^Use Model$/i));
      await screen.findByText(
        /^Select the Datasets and AI Model to be tested$/i
      );
      await screen.findByText(/^pickle_scikit_bc_compas.sav$/i);

      // Enter Test Arguments
      const algoTestArgsCard = container.querySelector(
        `div[id="algocard-${MOCK_ALGO_ID}"]`
      ) as HTMLElement;
      expect(algoTestArgsCard).not.toBeNull();
      await screen.findByText(/^Invalid Arguments$/i);
      const openBtn = algoTestArgsCard.querySelector(
        '.aivBase-button'
      ) as HTMLButtonElement;
      expect(openBtn).not.toBeNull();
      await user.click(openBtn);
      await screen.findByText(/^Sensitive Feature Names$/i);
      const sensitiveFeatureInput0 = container.querySelector(
        'input[name="root_sensitive_feature_0"]'
      ) as HTMLElement;
      await user.type(sensitiveFeatureInput0, 'race_code');
      await user.click(await screen.findByText(/^OK$/i));
      expect(
        await screen.queryByText(/^Sensitive Feature Names$/i)
      ).not.toBeInTheDocument();
      expect(
        await screen.queryByText(/^Invalid Arguments$/i)
      ).not.toBeInTheDocument();

      // Fill Input Block
      const iblockCard = container.querySelector(
        `div[id="ibcard-${MOCK_IBLOCK_ID}"]`
      ) as HTMLElement;
      expect(iblockCard).not.toBeNull();
      const openIblockBtn = iblockCard.querySelector(
        '.aivBase-button'
      ) as HTMLButtonElement;
      expect(openIblockBtn).not.toBeNull();
      await user.click(openIblockBtn);
      await screen.findByText(/^My MDX Input Block$/i);
      const fNameInput = container.querySelector('#fname') as HTMLInputElement;
      const lNameInput = container.querySelector('#lname') as HTMLInputElement;
      await user.type(fNameInput, 'John');
      await user.type(lNameInput, 'Doe');
      await user.click(await screen.findByText(/^OK$/i));
      expect(await screen.findByText(/^Invalid Inputs$/i)).toBeInTheDocument();
      await user.click(openIblockBtn);
      const bioInput = container.querySelector(
        'textarea'
      ) as HTMLTextAreaElement;
      await user.type(bioInput, 'Test input block');
      await user.click(await screen.findByText(/^OK$/i));
      await waitFor(async () => {
        expect(
          await screen.queryByTitle(/^Invalid Inputs$/i)
        ).not.toBeInTheDocument();
      });

      // click save
      const saveBtn = container.querySelector(
        'svg[data-testid="SaveIcon"]'
      ) as HTMLElement;
      await user.click(saveBtn);
      await screen.findByText(/^Project saved$/i);

      expect(container.querySelector('#inputsLeftpanel')).toMatchSnapshot(
        'leftPanel progress tracker AFTER valid inputs'
      );

      // Popup generate report confirmation
      await user.click(await screen.findByText(/^Next$/i));
      await screen.findByText(/^Confirm Generate Report$/i);
      expect(container.querySelector('#aivModal')).toHaveTextContent(
        'Fairness Metrics Toolbox for Regression'
      );
      await user.click(await screen.findByText(/^CANCEL$/i));
      expect(await screen.queryByText(/^Confirm Generate Report$/i)).toBeNull();
      await user.click(await screen.findByText(/^Next$/i));
      await screen.findByText(/^Confirm Generate Report$/i);
      await user.click(await screen.findByText(/^PROCEED$/i));
    }, 10000);
  });
});
