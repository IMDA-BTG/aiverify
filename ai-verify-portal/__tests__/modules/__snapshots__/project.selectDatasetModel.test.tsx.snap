// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Flow - Select Dataset And Model Datasets and Model Selection Screen should render Datasets / Model selection screen: Datasets/Model-Selection-Screen 1`] = `
<div
  class="layoutContentArea"
>
  <div
    class="scrollContainer"
  >
    <div
      class="container__limits"
    >
      <div
        class="inputsLayout"
      >
        <div
          class="stickyInputsLeftpanel"
          id="inputsLeftpanel"
        >
          <div
            class="projectDetail"
          >
            <h4>
              Project
            </h4>
            <h3>
              Test Project
            </h3>
          </div>
          <div
            class="root list__expanded"
            style="margin-top: 10px;"
          >
            <div
              class="header"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrowIcon css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m7 10 5 5 5-5z"
                />
              </svg>
              <div
                class="headerContent"
              >
                <div
                  class="headerChild"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DirectionsRunIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13.49 5.48c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-3.6 13.9 1-4.4 2.1 2v6h2v-7.5l-2.1-2 .6-3c1.3 1.5 3.3 2.5 5.5 2.5v-2c-1.9 0-3.5-1-4.3-2.4l-1-1.6c-.4-.6-1-1-1.7-1-.3 0-.5.1-.8.1l-5.2 2.2v4.7h2v-3.4l1.8-.7-1.6 8.1-4.9-1-.4 2 7 1.4z"
                    />
                  </svg>
                  <span
                    style="color: rgb(103, 103, 103); font-weight: 500;"
                  >
                    Tests Arguments
                  </span>
                </div>
              </div>
            </div>
            <div
              class="list"
            >
              <div
                class="item"
              >
                <div
                  class="itemWrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ErrorIcon"
                    focusable="false"
                    style="color: rgb(247, 57, 57); font-size: 29px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
                    />
                  </svg>
                  <div
                    class="itemText"
                  >
                    Fairness Metrics Toolbox for Regression
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="root list__expanded"
            style="margin-top: 10px;"
          >
            <div
              class="header"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrowIcon css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m7 10 5 5 5-5z"
                />
              </svg>
              <div
                class="headerContent"
              >
                <div
                  class="headerChild"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DynamicFormIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 20v-9h-2V4h7l-2 5h2l-5 11zm-2-7v7H4c-1.1 0-2-.9-2-2v-3c0-1.1.9-2 2-2h11zm-8.75 2.75h-1.5v1.5h1.5v-1.5zM13 4v7H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2h9zM6.25 6.75h-1.5v1.5h1.5v-1.5z"
                    />
                  </svg>
                  <span
                    style="color: rgb(103, 103, 103); font-weight: 500;"
                  >
                    Input Blocks Progress
                  </span>
                </div>
              </div>
            </div>
            <div
              class="list"
            >
              <div
                class="iBlockStepperWrapper"
              >
                <div
                  style="margin-top: 10px; margin-bottom: 10px;"
                >
                  <div
                    class="MuiStepper-root MuiStepper-vertical css-9tmuzo-MuiStepper-root"
                    style="text-align: center;"
                  >
                    <div
                      class="MuiStep-root MuiStep-vertical css-8wcvy5-MuiStep-root"
                    >
                      <span
                        class="MuiStepLabel-root MuiStepLabel-vertical css-14sza3e-MuiStepLabel-root"
                      >
                        <span
                          class="MuiStepLabel-iconContainer Mui-active css-vnkopk-MuiStepLabel-iconContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-1p22c1a-MuiSvgIcon-root"
                            data-testid="CircleIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                        >
                          <span
                            class="MuiStepLabel-label Mui-active css-1hv8oq8-MuiStepLabel-label"
                          >
                            <div
                              style="cursor: pointer; color: rgb(103, 103, 103); font-weight: 600;"
                            >
                              Mock Input Block 1
                            </div>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="inputsRightpanel"
          id="inputsRightPanel"
        >
          <div>
            <div
              class="inputSection"
            >
              <div
                class="sectionHeading"
              >
                <h3
                  class="screenHeading"
                >
                  Select the Datasets and AI Model to be tested
                </h3>
                <p
                  class="headingDescription"
                >
                  Please select the datasets required for report generation followed by the AI Model to be tested
                </p>
              </div>
              <div
                class="inputCard"
              >
                <div
                  class="inputDescription"
                  style="width: 350px;"
                >
                  <h4>
                    Testing Dataset
                  </h4>
                  <p>
                    Select the testing dataset to be used for the tests to run.
                  </p>
                </div>
                <div
                  class="divider"
                />
                <div
                  class="inputMain"
                >
                  <button
                    class="aivBase-button aivBase-button--outlined aivBase-button--small"
                    id="chooseTestDatasetBtn"
                  >
                    Choose Dataset
                  </button>
                </div>
              </div>
              <div
                class="inputCard"
              >
                <div
                  class="inputDescription"
                  style="width: 350px;"
                >
                  <h4>
                    Ground Truth Dataset
                  </h4>
                  <p>
                    Select the dataset containing the ground truth. The testing dataset can be used.
                  </p>
                </div>
                <div
                  class="divider"
                />
                <div
                  class="inputMain"
                >
                  <button
                    class="aivBase-button aivBase-button--outlined aivBase-button--small"
                    id="chooseGroundTruthBtn"
                  >
                    Choose Dataset
                  </button>
                </div>
              </div>
              <div
                class="inputCard"
              >
                <div
                  class="inputDescription"
                  style="width: 350px;"
                >
                  <h4>
                    AI Model
                  </h4>
                  <p>
                    Select the AI Model to be tested. Model Upload and API Connection available.
                  </p>
                </div>
                <div
                  class="divider"
                />
                <div
                  class="inputMain"
                >
                  <button
                    class="aivBase-button aivBase-button--outlined aivBase-button--small"
                  >
                    Choose Model
                  </button>
                </div>
              </div>
            </div>
            <div
              class="inputSection"
            >
              <div
                class="sectionHeading"
              >
                <h3
                  class="screenHeading"
                >
                  Provide Test Arguments
                </h3>
                <p
                  class="headingDescription"
                >
                  Open each test argument to provide the arguments required to run the tests and generate the report
                </p>
              </div>
              <div
                class="inputCard"
                id="algocard-aiverify.test.mock_test_plugin1:mock_algo1"
              >
                <div
                  class="inputDescription"
                  style="width: 350px;"
                >
                  <h4>
                    Fairness Metrics Toolbox for Regression
                  </h4>
                </div>
                <div
                  class="divider"
                />
                <div
                  class="inputMain"
                >
                  <p>
                    This is a mock description of mock algo 1
                  </p>
                  <div
                    class="buttonRow"
                  >
                    <button
                      class="aivBase-button aivBase-button--primary aivBase-button--small"
                    >
                      Open
                    </button>
                    <div
                      class="errorMsg"
                    >
                      Invalid Arguments
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="inputSection"
          >
            <div
              class="sectionHeading"
            >
              <h3
                class="screenHeading"
              >
                Complete Input Blocks
              </h3>
              <p
                class="headingDescription"
              >
                Open each input block to provide the information required
              </p>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="inputCard"
                id="ibcard-aiverify.test.mock_test_plugin1:mock_iblock1"
              >
                <div
                  class="inputDescription"
                  style="width: 350px;"
                >
                  <h4>
                    Mock Input Block 1
                  </h4>
                  <div
                    class="inputSummary"
                  />
                </div>
                <div
                  class="divider"
                />
                <div
                  class="inputMain"
                >
                  <p />
                  <div
                    class="buttonRow"
                  >
                    <button
                      class="aivBase-button aivBase-button--primary aivBase-button--small"
                    >
                      Open
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Project Flow - Select Dataset And Model Datasets and Model Selection Screen should select Datasets, Ground truth, capture Test arguments, Input Blocks and call generate report endpoints: leftPanel progress tracker AFTER valid inputs 1`] = `
<div
  class="stickyInputsLeftpanel"
  id="inputsLeftpanel"
>
  <div
    class="projectDetail"
  >
    <h4>
      Project
    </h4>
    <h3>
      Test Project
    </h3>
  </div>
  <div
    class="root list__expanded"
    style="margin-top: 10px;"
  >
    <div
      class="header"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrowIcon css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m7 10 5 5 5-5z"
        />
      </svg>
      <div
        class="headerContent"
      >
        <div
          class="headerChild"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DirectionsRunIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13.49 5.48c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-3.6 13.9 1-4.4 2.1 2v6h2v-7.5l-2.1-2 .6-3c1.3 1.5 3.3 2.5 5.5 2.5v-2c-1.9 0-3.5-1-4.3-2.4l-1-1.6c-.4-.6-1-1-1.7-1-.3 0-.5.1-.8.1l-5.2 2.2v4.7h2v-3.4l1.8-.7-1.6 8.1-4.9-1-.4 2 7 1.4z"
            />
          </svg>
          <span
            style="color: rgb(103, 103, 103); font-weight: 500;"
          >
            Tests Arguments
          </span>
        </div>
      </div>
    </div>
    <div
      class="list"
    >
      <div
        class="item"
      >
        <div
          class="itemWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckCircleIcon"
            focusable="false"
            style="color: rgb(112, 47, 138); font-size: 28px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          <div
            class="itemText"
          >
            Fairness Metrics Toolbox for Regression
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="root list__expanded"
    style="margin-top: 10px;"
  >
    <div
      class="header"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrowIcon css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m7 10 5 5 5-5z"
        />
      </svg>
      <div
        class="headerContent"
      >
        <div
          class="headerChild"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DynamicFormIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 20v-9h-2V4h7l-2 5h2l-5 11zm-2-7v7H4c-1.1 0-2-.9-2-2v-3c0-1.1.9-2 2-2h11zm-8.75 2.75h-1.5v1.5h1.5v-1.5zM13 4v7H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2h9zM6.25 6.75h-1.5v1.5h1.5v-1.5z"
            />
          </svg>
          <span
            style="color: rgb(103, 103, 103); font-weight: 500;"
          >
            Input Blocks Progress
          </span>
        </div>
      </div>
    </div>
    <div
      class="list"
    >
      <div
        class="iBlockStepperWrapper"
      >
        <div
          style="margin-top: 10px; margin-bottom: 10px;"
        >
          <div
            class="MuiStepper-root MuiStepper-vertical css-9tmuzo-MuiStepper-root"
            style="text-align: center;"
          >
            <div
              class="MuiStep-root MuiStep-vertical Mui-completed css-8wcvy5-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-vertical css-14sza3e-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-active Mui-completed css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-1p22c1a-MuiSvgIcon-root"
                    data-testid="CheckCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active Mui-completed css-1hv8oq8-MuiStepLabel-label"
                  >
                    <div
                      style="cursor: pointer; color: rgb(103, 103, 103); font-weight: 600;"
                    >
                      Mock Input Block 1
                    </div>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
